test/associations_test.rb

```diff
@@ -25,42 +25,60 @@ def setup
   end
 
   def test_has_many_associations
-    tea = Tea.new("Name" => "Dong Ding", "Brews" => ["rec2"])
+    tea = Tea.new("Name" => "Dong Ding", "Brews" => ["rec2", "rec1"])
 
-    record = Brew.new("Name" => "Good brew")
-    stub_find_request(record, id: "rec2", table: Brew)
+    brews = [
+      { "id" => "rec2", "Name" => "Good brew" },
+      { "id" => "rec1", "Name" => "Decent brew" }
+    ]
+    stub_request(brews, table: Brew)
 
-    assert_equal 1, tea["Brews"].size
-    assert_kind_of Airrecord::Table, tea["Brews"].first
-    assert_equal "rec2", tea["Brews"].first.id
+    assert_equal 2, tea.brews.size
+    assert_kind_of Airrecord::Table, tea.brews.first
+    assert_equal "rec1", tea.brews.first.id
   end
 
   def test_belongs_to
     brew = Brew.new("Name" => "Good Brew", "Tea" => ["rec1"])
     tea = Tea.new("Name" => "Dong Ding", "Brews" => ["rec2"])
     stub_find_request(tea, table: Tea, id: "rec1")
 
-    assert_equal "rec1", brew["Tea"].id
+    assert_equal "rec1", brew.tea.id
+  end
+
+  def test_build_association_from_strings
```

Comment:
Let's add a test case for `["Brews"] = [<Brew>, <Brew>]` — this should either (1) raise ("please don't do this"), (2) use the association setter, or (3) call `map(&:id)`. Which one do you think is more natural?

Reply:
I think one of the advantages of this new association API is that associated fields no longer behave differently than regular fields under the hood. They become just fields, and they no longer need to be tracked separately and treated exceptionally.

I can't think of a way of implementing (1) or (2) without needing to track association fields separately again.

I propose we either

**(a)** Do nothing. Airtable's API already returns a semi-helpful error message in this case:
```ruby
tea["Brews"] = [<Brew>]
tea.save
#=> Airrecord::Error (HTTP 422: INVALID_RECORD_ID: Value "#<Brew:0x00007f9080147f18>" is not a valid record ID.)
```

or

**(b)** Override `#to_s` to get the behavior you suggested in (3).

If we define `#to_s` like this...
```ruby
def to_s; id; end
```
... then we'll get a map of ids in the HTTP request without needing to track association fields separately.

I think (a) is probably a better choice, both because I'd prefer to raise an error in this scenario, and because overriding `to_s` makes me nervous outside of very particular contexts. What do you think?

Reply:
Good with me. I love APIs with good errors that put less onus on the wrappers. 👍